XPlat/Shopify/iOS/source/Shopify.iOS/ApiDefinition.cs

```diff
@@ -0,0 +1,1964 @@
+		IntPtr Constructor (string cartToken);
+
+		// -(BOOL)hasToken;
+		[Export ("hasToken")]
+		bool HasToken { get; }
+
+		// @property (readonly, copy, nonatomic) NSNumber * orderId __attribute__((deprecated("Available on the BUYOrder object")));
+		[Export ("orderId", ArgumentSemantic.Copy)]
+		NSNumber OrderId { get; }
+
+		// @property (readonly, nonatomic, strong) NSURL * orderStatusURL __attribute__((deprecated("Available on the BUYOrder object")));
+		[Export ("orderStatusURL", ArgumentSemantic.Strong)]
+		NSUrl OrderStatusUrl { get; }
+	}
+
+	// @interface BUYCheckoutAttribute : BUYObject <BUYSerializable>
+	[BaseType (typeof(Object), Name = "BUYCheckoutAttribute")]
+	interface CheckoutAttribute : Serializable
+	{
+		// @property (nonatomic, strong, nonnull) NSString *name;
+		[Export ("name", ArgumentSemantic.Strong)]
+		string Name { get; set; }
+
+		// @property (nonatomic, strong, nonnull) NSString *value;
+		[Export ("value", ArgumentSemantic.Strong)]
+		string Vitle { get; set; }
+	}
+
+	// @interface BUYShippingRate : BUYObject <BUYSerializable>
+	[BaseType (typeof(Object), Name = "BUYShippingRate")]
+	interface ShippingRate : Serializable
+	{
+		// @property (readonly, nonatomic, strong) NSString * shippingRateIdentifier;
+		[Export ("shippingRateIdentifier", ArgumentSemantic.Strong)]
+		string ShippingRateIdentifier { get; }
+
+		// @property (readonly, nonatomic, strong) NSString * title;
+		[Export ("title", ArgumentSemantic.Strong)]
+		string Title { get; }
+
+		// @property (readonly, nonatomic, strong) NSDecimalNumber * price;
+		[Export ("price", ArgumentSemantic.Strong)]
+		NSDecimalNumber Price { get; }
+
+		// TODO: DeliveryRange might be NSDate[]
+
+		// @property (readonly, nonatomic, strong) NSArray * deliveryRange;
+		[Export ("deliveryRange", ArgumentSemantic.Strong)]
+		NSObject[] DeliveryRange { get; }
+
+
+		// @interface ApplePay (BUYShippingRate)
+
+		// +(NSArray<PKShippingMethod *> * _Nonnull)buy_convertShippingRatesToShippingMethods:(NSArray<BUYShippingRate *> * _Nonnull)rates;
+		[Static]
+		[Export ("buy_convertShippingRatesToShippingMethods:")]
+		PKShippingMethod[] ApplePayConvertShippingRatesToShippingMethods (ShippingRate[] rates);
+
+	}
+
+	// @interface ApplePay (BUYCheckout)
+	[Category]
+	[BaseType (typeof(Checkout), Name = "BUYCheckout")]
+	interface CheckoutApplePayExtensions
+	{
+		// -(NSArray<PKPaymentSummaryItem *> * _Nonnull)buy_summaryItems;
+		[Export ("buy_summaryItems")]
+		PKPaymentSummaryItem[] ApplePaySummaryItems ();
+
+		// -(NSArray<PKPaymentSummaryItem *> * _Nonnull)buy_summaryItemsWithShopName:(NSString * _Nullable)shopName;
+		[Export ("buy_summaryItemsWithShopName:")]
+		PKPaymentSummaryItem[] ApplePaySummaryItems ([NullAllowed] string shopName);
+	}
+
+	// @interface BUYApplePayHelpers : NSObject <PKPaymentAuthorizationViewControllerDelegate>
+	[BaseType (typeof(NSObject), Name = "BUYApplePayHelpers")]
+	interface ApplePayHelpers : IPKPaymentAuthorizationViewControllerDelegate
```

Comment:
I would suggest to add [DisableDefaultCtor] attribute because Client and Checkout parameters cannot be set after instantiating.